src/python/pants/core/goals/test.py

```diff
@@ -388,6 +410,45 @@ class Test(Goal):
     __test__ = False
 
 
+@rule_helper
+async def _run_debug_tests(
+    test_subsystem: TestSubsystem,
+) -> Test:
+    targets_to_valid_field_sets = await Get(
+        TargetRootsToFieldSets,
+        TargetRootsToFieldSetsRequest(
+            TestFieldSet,
+            goal_description="`test --debug`",
```

Comment:
Should be conditional on `--debug` vs `--debug-adapter`.

Reply:
(FWIW `--debug-adapter` implies `--debug` so that was my middle-ground. Doesn't hurt to ternary though